tests/AssertTest.php

```diff
@@ -495,6 +501,8 @@ public function getTests()
             array('isList', array(array(false)), true),
             array('isList', array(array(array(1), array(2))), true),
             array('isList', array(array(array('foo' => 'bar'), array('baz' => 'tab'))), true),
+            array('isList', array($nanList), true),
+            array('isList', array($normalList), true),
```

Comment:
I wonder if `nonEmptyList` should match as well, not sure if it does when `NAN` is passed. 
But TBH, I've never seen any code which worked with `NAN` so this is a very niche edge case which I would not miss if its not covered. 🤙🏼